src/epgsql_sock.erl

```diff
@@ -407,13 +435,22 @@ command_next(#state{current_cmd = PrevCmd,
                         results = []}
     end.
 
-
 setopts(#state{mod = Mod, sock = Sock}, Opts) ->
     case Mod of
         gen_tcp -> inet:setopts(Sock, Opts);
         ssl     -> ssl:setopts(Sock, Opts)
     end.
 
+-spec get_socket_active(pg_sock()) -> epgsql:socket_active().
+get_socket_active(#state{handler = H}) when H =/= on_replication ->
+    true;
+get_socket_active(#state{connect_opts = #{socket_active := Active}}) ->
```

Comment:
If you do this:

```suggestion
get_socket_active(#state{handler =:= on_replication, connect_opts = #{socket_active := Active}}) ->
```

and assume `connect_opts` is always map, you may drop the 1st and 3rd clauses?